opennms-alarms/daemon/src/main/java/org/opennms/netmgt/alarmd/AlarmPersisterImpl.java

```diff
@@ -154,74 +155,87 @@ private void reduceEvent(OnmsEvent e, OnmsAlarm alarm, Event event) {
         // Always set these
         alarm.setLastEvent(e);
         alarm.setLastEventTime(e.getEventTime());
-        alarm.setCounter(alarm.getCounter() + 1);
 
-        if (!event.getAlarmData().hasUpdateFields()) {
+        AlarmData alarmData = event.getAlarmData();
+        if (alarmData != null) {
 
-            //We always set these even if there are not update fields specified
-            alarm.setLogMsg(e.getEventLogMsg());
-        } else {
-            for (UpdateField field : event.getAlarmData().getUpdateFieldList()) {
-                String fieldName = field.getFieldName();
-
-                //Always set these, unless specified not to, in order to maintain current behavior
-                if (fieldName.equalsIgnoreCase("LogMsg") && !field.isUpdateOnReduction()) {
-                    continue;
+            Integer newAlarmType = alarmData.getAlarmType();
+            Integer currentAlarmType = alarm.getAlarmType();
+            if (newAlarmType != null && currentAlarmType != null) {
+                if (!newAlarmType.equals(OnmsAlarm.RESOLUTION_TYPE)) {
+                    alarm.setCounter(alarm.getCounter() + 1);
                 } else {
-                    alarm.setLogMsg(e.getEventLogMsg());
+                    alarm.setAlarmType(newAlarmType);
```

Comment:
It looks like an alarm can be updated from type 1 to type 2, but it won't get updated back to type 1 if the problem reoccurs.

Reply:
Fixed in latest commit.